tests/epyccel/test_arrays_multiple_assignments.py

```diff
@@ -165,6 +166,59 @@ def f(c):
     c = np.random.random()
     assert f(c) == g(c)
 
+#==============================================================================
+def test_Reassign_to_Target(language):
+
+    def f():
+        import numpy as np
+        x = np.zeros((3, 7), dtype=int)
+        c = x
+        x = np.ones ((4, 5), dtype=int)
+        return c.sum()
+
+     # Initialize singleton that stores Pyccel errors
+    errors = Errors()
+
+    # epyccel should raise an Exception
+    with pytest.raises(PyccelSemanticError):
+        epyccel(f, language=language)
+
+    # Check that we got exactly 1 Pyccel error
+    assert errors.has_errors() == 1
+    assert errors.num_messages() == 1
+
+    # Check that the error is correct
+    error_info = [*errors.error_info_map.values()][0][0]
+    assert error_info.symbol  == 'x'
+    assert error_info.message == ARRAY_ALREADY_IN_USE
+
+#==============================================================================
+def test_Assign_Between_Allocatables(language):
+
+    def f():
+        import numpy as np
+        x = np.zeros((3, 7), dtype=int)
+        y = np.ones ((4, 5), dtype=int)
+        x = y
+        x[0][0] = 1
+        return y.sum()
+
+     # Initialize singleton that stores Pyccel errors
+    errors = Errors()
+
+    # epyccel should raise an Exception
+    with pytest.raises(PyccelSemanticError):
+        epyccel(f, language=language)
+
+    # Check that we got exactly 1 Pyccel error
+    assert errors.has_errors() == 1
+    assert errors.num_messages() == 1
+
+    # Check that the error is correct
+    error_info = [*errors.error_info_map.values()][0][0]
+    assert error_info.symbol  == 'x'
+    assert error_info.message == ASSIGN_ARRAYS_ONE_ANOTHER
+
```

Comment:
Can the tests in this file be activated for C now?

Reply:
we are using the NumpySum / NumpyRand node are not implemented in C yet

Reply:
So then none of this is tested at all?

Reply:
they tests that I added are for raising errors in the semantic stage

Reply:
I thought `test_arrays_multiple_assignments.py` tested for an error raised in the semantic stage? Can't we activate those tests? The codegen stage isn't carried out if the semantic step raises an error

Reply:
Could we activate the tests in `test_pointers.py`?

Reply:
i need to remove the shape function from the tests/epyccel/modules/pointers.py

Reply:
the shape func is still not implemented in C

Reply:
In that case could you add some new more basic tests that should work already

Reply:
for test_pointers.py we can actually activate it now I thought that we were using the shape method node numpyshape

Reply:
We are using the numpy shape method. But in pyccel that is implemented by returning either the known value of the shape, or an instance of `PyccelArraySize`

Reply:
> the tests that I added are for raising errors in the semantic stage

So the tests that you added should work for any language

Reply:
sorry, I just got confused, I activated the tests for pointers and add a test for the case of the reassign of a pointer